modules/javafx.graphics/src/main/java/com/sun/javafx/css/BinarySerializer.java

```diff
@@ -103,7 +103,11 @@ private static CompoundSelector readCompoundSelector(DataInputStream is, String[
              * before attempting to read what is known to be a simple selector.
              */
 
-            is.readByte();
+            byte type = is.readByte();
+
+            if (type != TYPE_SIMPLE) {
+                throw new IllegalStateException("Expected compound selector to consist of simple selectors only, but found type: " + type);
```

Comment:
minor: the output will be a signed decimal.  would it be better if this can be formatted as hex?

`String.format("Expected compound selector to consist of TYPE_SIMPLE only, but found type: 0x%02X", type)`

Reply:
I don't think that would matter much.  This error will only ever be interpreted by programmers, I'm sure they'll be able to understand what's going on.

Reply:
Agreed. This should never happen, and if it does, the difference between a hex and decimal printout for the erroneous value wouldn't matter.

Reply:
I disagree, especially since the target audience is developers.  When this happens and the dev needs to diagnose, hex is better: some things become immediately obvious, and some can be found in the binary viewer which is always in hex.  I don't understand why you resist.

Will re-approve if you decide to fix it.